packages/mui-material/src/Avatar/Avatar.js

```diff
@@ -46,15 +46,17 @@ const AvatarRoot = styled('div', {
   overflow: 'hidden',
   userSelect: 'none',
   ...(ownerState.variant === 'rounded' && {
-    borderRadius: theme.shape.borderRadius,
+    borderRadius: (theme.vars || theme).shape.borderRadius,
   }),
   ...(ownerState.variant === 'square' && {
     borderRadius: 0,
   }),
   ...(ownerState.colorDefault && {
-    color: theme.palette.background.default,
+    color: (theme.vars || theme).palette.background.default,
     backgroundColor:
-      theme.palette.mode === 'light' ? theme.palette.grey[400] : theme.palette.grey[600],
+      theme.palette.mode === 'light'
+        ? (theme.vars || theme).palette.grey[400]
+        : (theme.vars || theme).palette.grey[600],
```

Comment:
Switching between tokens will cause a flicker for SSR application. We have to come up with a new variable in the theme palette.

Reply:
@siriwatknp How or where do we do this?

Reply:
We have concluded on https://github.com/mui/material-ui/issues/32049#issuecomment-1124601805